vote shop fix + multidimensional homes

Rg-Winters opened the pull request:
Fixes issue with player losing credits but not receiving items they purchase in the vote shop
adds an error message for problems with purchases
adds a purchase success message for a successful purchase

Rg-Winters committed:
vote shop fix

Olexorus commented:
Looks good, but there's no need to create a separate scoreboard objective, since the value is only needed in the function and doesn't need to be stored afterwards.
I'd like to keep the amount of scoreboard objectives low, that's why there are temp_1 to 3 for things like these, please change it to use that.

Rg-Winters committed:
Allows for multi dimensional home setting!

Rg-Winters committed:
new change fix 1

Rg-Winters committed:
change 2

Rg-Winters committed:
change 3

Rg-Winters committed:
change 3

Rg-Winters committed:
change 4

Rg-Winters committed:
change 5

Rg-Winters commented:
alright I added multi dimensional homes, and removed that unnecessary variable per your request. I did add a couple of new variables for multidimensional homes, mainly a variable that stores your home's dimension, and a new temp variable so I could relay the dimension for the tp as well as the coordinates of a home. Let me know if you have a different/better idea on how I implemented it.
